src/java.base/share/classes/sun/security/util/Hybrid.java

```diff
@@ -0,0 +1,435 @@
+            this.leftlen = leftPublicLength(left);
+            this.leftname = left;
+            this.rightname = right;
+        }
+
+        @Override
+        protected PublicKey engineGeneratePublic(KeySpec keySpec)
+                throws InvalidKeySpecException {
+            if (keySpec instanceof RawKeySpec rks) {
+                byte[] key = rks.getKeyArr();
+                byte[] leftKeyBytes = Arrays.copyOfRange(key, 0, leftlen);
+                byte[] rightKeyBytes = Arrays.copyOfRange(key, leftlen,
+                        key.length);
+                PublicKey leftKey, rightKey;
+
+                try {
+                    if (APS.isGenericEC(leftname)) {
+                        var curve = CurveDB.lookup(leftname);
+                        var ecSpec = new ECPublicKeySpec(
+                                ECUtil.decodePoint(leftKeyBytes,
+                                curve.getCurve()), curve);
+                        leftKey = left.generatePublic(ecSpec);
+                    } else if (leftname.startsWith("ML-KEM")) {
+                        try {
+                            leftKey = left.generatePublic(new RawKeySpec(
+                                    leftKeyBytes));
+                        } catch (Exception e) {
+                            leftKey = left.generatePublic(new X509EncodedKeySpec(
+                                    leftKeyBytes));
+                        }
+                    } else {
+                        throw new InvalidKeySpecException("Unsupported left" +
+                                " algorithm" + leftname);
+                    }
+
+                    if (APS.isXDH(rightname)) {
+                        ArrayUtil.reverse(rightKeyBytes);
+                        var xecSpec = new XECPublicKeySpec(
+                                new NamedParameterSpec(rightname),
+                                new BigInteger(1, rightKeyBytes));
+                        rightKey = right.generatePublic(xecSpec);
+                    } else if (rightname.startsWith("ML-KEM")) {
+                        try {
+                            rightKey = right.generatePublic(new RawKeySpec(
+                                    rightKeyBytes));
+                        } catch (Exception e) {
+                            rightKey = right.generatePublic(new X509EncodedKeySpec(
+                                    rightKeyBytes));
+                        }
+                    } else {
+                        throw new InvalidKeySpecException("Unsupported right" +
+                                " algorithm: " + rightname);
+                    }
+
+                    return new PublicKeyImpl("Hybrid", leftKey, rightKey);
+                } catch (Exception e) {
+                    throw new InvalidKeySpecException("Failed to decode hybrid" +
+                            " key", e);
+                }
+            }
+
+            throw new InvalidKeySpecException(keySpec.toString());
```

Comment:
Please check null keySpec, and  `throw new InvalidKeySpecException("keySpec must not be null")`.

For non-null keySpec, I may use:
`throw new InvalidKeySpecException(keySpec.getClass().getName() + " not supported.");
`

Reply:
Added null `keySpec` check, and updated `InvalidKeySpecException` message.